services/galley/src/Galley/API/Util.hs

```diff
@@ -193,53 +193,67 @@ acceptOne2One usr conv conn = case Data.convType conv of
         "Connect conversation with more than 2 members: "
           <> LT.pack (show cid)
 
-isBot :: Member -> Bool
+isBot :: InternalMember a -> Bool
 isBot = isJust . memService
 
-isMember :: Foldable m => OpaqueUserId -> m Member -> Bool
-isMember u = isJust . find ((u ==) . makeIdOpaque . memId)
+isMember :: (Eq a, Foldable m) => a -> m (InternalMember a) -> Bool
+isMember u = isJust . find ((u ==) . memId)
 
-findMember :: Data.Conversation -> UserId -> Maybe Member
+findMember :: Data.Conversation -> MappedOrLocalId Id.U -> Maybe Member
 findMember c u = find ((u ==) . memId) (Data.convMembers c)
 
 botsAndUsers :: Foldable t => t Member -> ([BotMember], [Member])
-botsAndUsers = foldr fn ([], [])
+botsAndUsers = foldMap fn
   where
-    fn m ~(bb, mm) = case newBotMember m of
-      Nothing -> (bb, m : mm)
-      Just b -> (b : bb, mm)
+    fn m = case memService m of
+      Just _ ->
+        -- TODO(mheinzel): we drop invalid bots here, which shouldn't happen
```

Comment:
This happens because Members can now be remote, but we still assume bots to be local (see definition of `BotMember`). If there is a remote bot member, this would just ignore it.